pkg/dns/util/util.go

```diff
@@ -116,3 +116,11 @@ func ValidateNameserverIpAndPort(nameServer string) (string, string, error) {
 func IsServiceIPSet(service *corev1.Service) bool {
 	return service.Spec.ClusterIP != corev1.ClusterIPNone && service.Spec.ClusterIP != ""
 }
+
+// GetClusterIPs returns IPs set for the service
+func GetClusterIPs(service *corev1.Service) []string {
+	if len(service.Spec.ClusterIPs) > 0 {
```

Comment:
Can we add some defensive code and flag gate here?



if ClusterIP does not appear in ClusterIPs, let's log an error default back to using ClusterIP only.

Reply:
Also -- add a test case for this..

Reply:
> Can we add some defensive code and flag gate here?

What do you mean by defensive code here?
As for the flag gate, should we introduce command line argument to toggle the behaviour? Curious in what cases would it be needed to explicitly ignore `service.Spec.ClusterIPs` if it's present in favor of `service.Spec.ClusterIP`?

> if ClusterIP does not appear in ClusterIPs, let's log an error default back to using ClusterIP only.

If I understand correctly, logging an error here would be sort of a backwards incompatible change, e.g. if ipv4-only cluster gets this version it would just start logging errors. 
But I agree that it would probably be useful to have debug/info logs to be able quickly identify the source used to created dns entries.

> Also -- add a test case for this..

I didn't add explicit test for this func as it's invoked and the behaviour is controlled in `TestUnnamedSinglePortService` and `TestNamedSinglePortService` tests